src/Arcus.EventGrid.Publishing/EventGridPublisher.cs

```diff
@@ -256,14 +304,29 @@ public async Task PublishManyAsync<TEvent>(IEnumerable<TEvent> events)
             return content;
         }
 
-        private async Task PublishContentToTopicAsync(HttpContent content)
+        private async Task PublishContentToTopicAsync(HttpContent content, string logSubject)
         {
-            using (HttpResponseMessage response = 
-                await _resilientPolicy.ExecuteAsync(() => SendAuthorizedHttpPostRequestAsync(content)))
+            using (DependencyMeasurement measurement = DependencyMeasurement.Start())
             {
-                if (!response.IsSuccessStatusCode)
+                var isSuccessful = false;
+                try
+                {
+                    using (HttpResponseMessage response =
+                        await _resilientPolicy.ExecuteAsync(() => SendAuthorizedHttpPostRequestAsync(content)))
+                    {
+                        isSuccessful = response.IsSuccessStatusCode;
+                        if (!response.IsSuccessStatusCode)
+                        {
+                            await ThrowApplicationExceptionAsync(response);
+                        }
+                    }
+                }
+                finally
                 {
-                    await ThrowApplicationExceptionAsync(response);
+                    if (_options.EnableDependencyTracking)
+                    {
+                        _logger.LogDependency("Azure Event Grid", dependencyData: logSubject ?? "<no-subject>", targetName: TopicEndpoint, isSuccessful: isSuccessful, measurement); 
```

Comment:
I would use the event type here for now, which is more valuable

Reply:
Ok, let's do that! Resolving.